profiling/src/internal/observation/observations.rs

```diff
@@ -7,43 +7,58 @@ use super::super::Sample;
 use super::trimmed_observation::{ObservationLength, TrimmedObservation};
 use crate::internal::Timestamp;
 use std::collections::HashMap;
+use crate::collections::identifiable::*;
+use lz4_flex::frame::FrameEncoder;
+use std::io::Write;
 
 struct NonEmptyObservations {
     aggregated_data: HashMap<Sample, TrimmedObservation>,
-    timestamped_data: Vec<TrimmedTimestampedObservation>,
+    compressed_timestamped_data: FrameEncoder<Vec<u8>>,
     obs_len: ObservationLength,
 }
 
-// Timestamp and TrimmedObservation are both 64bit values
-// Using a 32 bit SampleId would still take 64 bits due to padding
-// So just put the Sample in here
-type TrimmedTimestampedObservation = (Sample, Timestamp, TrimmedObservation);
-
 #[derive(Default)]
 pub struct Observations {
     inner: Option<NonEmptyObservations>,
 }
 
 /// Public API
 impl Observations {
-    pub fn add(&mut self, sample: Sample, timestamp: Option<Timestamp>, values: Vec<i64>) {
-        if let Some(inner) = &self.inner {
-            inner.obs_len.assert_eq(values.len());
+    pub fn init_timeline(&mut self, values_size: usize) {
+        if let Some(_inner) = &self.inner {
+            panic!("Should never happen!");
         } else {
+            // // Create buffer, pre-touch all elements to avoid any laziness
+            let mut timestamped_data_buffer: Vec<u8> = vec![0; 1_048_576];
```

Comment:
Make this a configurable constant

Reply:
Do you mind a build-time constant? Or something that can be configured dynamically by the library?

Reply:
buildtime is fine

Reply:
Added in 6157a90e479f67c22f04c3bab214588be4801fed